libpod/container_internal.go

```diff
@@ -946,12 +946,21 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 	}
 
 	if c.config.User != "" {
+		var (
+			uid uint32
+			gid uint32
+		)
 		if !c.state.Mounted {
 			return nil, errors.Wrapf(ErrCtrStateInvalid, "container %s must be mounted in order to translate User field", c.ID())
 		}
-		uid, gid, err := chrootuser.GetUser(c.state.Mountpoint, c.config.User)
-		if err != nil {
-			return nil, err
+		// if user is an int, don't look it up in /etc/passwd, just set the ProcessUID to the int given
+		if intUser, err := strconv.Atoi(c.config.User); err == nil {
+			uid = uint32(intUser)
```

Comment:
In this case do you end up with a gid==0?  We should be looking up the UID in the /etc/passwd in the container to get its defeault UID, if there is not entry GID=UID would be a better default then sticking the UID=0 into the user from a secuity point of view.

Reply:
uid, gid, err = chrootuser.GetUserByUID(uid, c.state.Mountpoint, c.config.User)

Reply:
I was following docker and it ends up with GID = 0 for these cases. Should I change it to GID == UID?

Reply:
Yikes.  I am not sure now.  From a security point of view group==0 is insecure.   Does docker look up the GID via UID?  I think we need at least this.

Reply:
Not sure if there is an exported function, but this is what you need.
```
 	_, gid, err := lookupGroupForUIDInContainer(rootdir, userid)
```

Reply:
Okay I see what docker is doing, so even if the --user flag is set to an int, they look up this uid in /etc/passwd, if it exists they get the GID based on the UID. For the case where it doesn't exist, the GID ends up being 0. So we should check /etc/passwd every time and just not fail if we can't find the UID.
```
➜  libpod git:(user) sudo docker run -it --rm --user 8 busybox id
uid=8(mail) gid=8(mail)
➜  libpod git:(user) sudo docker run -it --rm --user 20 busybox id
uid=20 gid=0(root)
```